plugins/user-settings/src/components/AuthProviders/ProviderSettingsItem.tsx

```diff
@@ -50,7 +57,16 @@ export const ProviderSettingsItem = (props: {
       .sessionState$()
       .subscribe((sessionState: SessionState) => {
         if (!didCancel) {
-          setSignedIn(sessionState === SessionState.SignedIn);
+          api
+            .getProfile({ optional: true })
+            .then((profileResponse: ProfileInfo | undefined) => {
+              if (sessionState === SessionState.SignedIn) {
+                setSignedIn(true);
+              } else {
+                setSignedIn(profileResponse !== undefined);
+              }
+              setProfile(profileResponse ?? {});
```

Comment:
this should probably be

```ts
if (sessionState === SessionState.SignedIn) {
  setSignedIn(true);
}
if (profileResponse) {
    setProfile(profileResponse);
}
```

no need to re-set the profile if nothing was given, and I'm not sure that ONLY getting a profile response should constitute being signed in.

Reply:
ahh good suggestion, I completely missed that

> I'm not sure that ONLY getting a profile response should constitute being signed in.

Is this just a thought, or is that a request for change? I would agree, and my assumption would be that there would be a `isSignedIn()` function, but I believe that is the current shortcoming that we are trying to solve, ie. we can check whether the user is signed in, only on one provider. I checked whether there would be some other ways to check whether user is logged in, but all of them create a pop-up for the user to log in if they are not, which I guess would be annoying.